test/spec_set_x_forwarded_proto_header.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+require_relative 'helper'
+
+
+describe Rack::SetXForwardedProtoHeader do
+  response = lambda {|e| [200, {}, []]  }
+
+  it "leaves the value of X_FORWARDED_PROTO intact if there is no vendor header passed in the request" do
+    vendor_forwarded_header = "not passed in the request"
+    env = Rack::MockRequest.env_for("/", "HTTP_X_FORWARDED_PROTO" => "http")
+
+    Rack::Lint.new(Rack::SetXForwardedProtoHeader.new(response, vendor_forwarded_header)).call env
+
+    env["HTTP_X_FORWARDED_PROTO"].must_equal "http"
+  end
+
+  it "returns early when there is no vendor header passed in the request" do
+    vendor_forwarded_header = "not passed in the request"
+    env = Rack::MockRequest.env_for("/", "FOO" => "bar")
+
+    header_middleware = Rack::SetXForwardedProtoHeader.new(response, vendor_forwarded_header)
+    # Patch to ensure we return early and do not call `copy_header_value`
+    def header_middleware.copy_header_value
+      raise NoMethodError, "should never be called when vendor_forwarded_header is not in the request"
+    end
+
+    Rack::Lint.new(header_middleware).call env
+
+    env["FOO"].must_equal "bar"
+  end
+
+
+  it "copies the value of the header to X-Forwarded-Proto" do
+    env = Rack::MockRequest.env_for("/", "HTTP_CLOUDFRONT_FORWARDED_PROTO" => "https")
+
+    Rack::Lint.new(Rack::SetXForwardedProtoHeader.new(response, "CloudFront-Forwarded-Proto")).call env
+
+    env["HTTP_X_FORWARDED_PROTO"].must_equal "https"
+  end
+
```

Comment:
Should add a spec to test when both the vendor header and `HTTP_X_FORWARDED_PROTO` are set, testing that the vendor header overwrites an existing `HTTP_X_FORWARDED_PROTO` header.

Reply:
✅